### ISSUE TITLE
Stanza include directory is not fully-resolved

### ISSUE THREAD

**Issue by tylanphear** [StanzaOrg/lbstanza-old#130]
This can be observed by running `stanza path` with a `.stanza` with an entry `install-dir = <path>` where `<path>` is reachable from the current directory. e.g:

```stanza
install-dir = "."
```
```
> ./stanza path
.
```

This one issue in isolation seems harmless enough, but has larger implications for other standard locations that are relative to the Stanza install-dir (e.g. the `StanzaIncludeDir`). For example, if you recreate the same setup as above, and compile any stanza file, the compiler output makes it clear that relative paths are being used, e.g:

```
> stanza foo.stanza -o foo -verbose
...
Call C compiler with arguments:
  "gcc"
  "temp520360337.s"
  "runtime\\driver.c"
  "-std=gnu99"
  "-lm"
  "-D"
  "PLATFORM_WINDOWS"
  "-I"
  "include"
  "-o"
  "test.exe"
```

Why is this an issue? Consider the case where `-external-dependencies <deps-file>` is provided on the same command line. In that case, a set of dependencies is emitted in the form

```
ccfiles :  
  "runtime\\driver.c"
ccflags :  
  "-std=gnu99"
  "-lm"
  "-D"
  "PLATFORM_WINDOWS"
  "-I"
  "include"
```

This accidentally encodes the location where the Stanza compiler was invoked into the final artifact, making it impossible for a consumer of this file to consume it without recreating the paths relative to the path where the compiler was invoked. Additionally, this only manifests in the above-described configuration. If the same invocation is performed from another directory, the following output is produced:

```
> STANZA_CONFIG="$HOME/lbstanza" stanza test.stanza -o test -verbose
...
Call C compiler with arguments:
  "gcc"
  "temp1895704451.s"
  "C:\\msys64\\home\\Tyler\\lbstanza\\runtime\\driver.c"
  "-std=gnu99"
  "-lm"
  "-D"
  "PLATFORM_WINDOWS"
  "-I"
  "C:\\msys64\\home\\Tyler\\lbstanza\\include"
  "-o"
  "test.exe"
```

The same is true with respect to the contents of the generated `<deps-file>`.

There is no real gain to having the compiler use relative paths over absolute paths, and more than a few downsides. One of the places that this rears its ugly head is in the JITX build. There, there is some post-processing in the `cmake-formatter` to clean up relative paths in external dependency files that point to the wrong path (e.g. `runtime/driver.c`).

Hence, I think the compiler should resolve `install-dir` into an absolute path and then form any standard Stanza paths (like `StanzaRuntimeDriver` and `StanzaIncludeDir`) relative to that absolute path.

**Comment by tylanphear** [comment]
The plot thickens. After looking at the relevant code (in `compiler/stz-config.stanza`), I realized we are already trying to resolve relative paths. Why then, are we seeing relative paths still being used? It appears that the above only occurs if I set my `STANZA_CONFIG` to a relative path (for example, in the above I used `STANZA_CONFIG="."`). Likely the problem is that `STANZA_CONFIG` needs to be resolved to an absolute path instead of using a relative path.
